cli/compose/types/types.go

```diff
@@ -260,6 +262,15 @@ type ServiceSecretConfig struct {
 	Mode   *uint32
 }
 
+// ServiceConfigObjConfig is the config obj configuration for a service
+type ServiceConfigObjConfig struct {
```

Comment:
I'm really worried about doing this.
They happen to share similar code because they currently provide vary similar functionality, but I consider this coincidental duplication.

I would rather see where/how secrets and configs take off before trying to de-dup.